packages/application-extension/src/index.ts

```diff
@@ -178,9 +180,12 @@ const opener: JupyterFrontEndPlugin<void> = {
 
         const file = decodeURIComponent(path);
         const ext = PathExt.extname(file);
-        app.restored.then(() => {
+        app.restored.then(async () => {
           // TODO: get factory from file type instead?
           if (ext === '.ipynb') {
+            // TODO: fix upstream?
+            await settingRegistry?.load('@jupyterlab/notebook-extension:panel');
+            await Promise.resolve();
```

Comment:
We should investigate if it is possible to fix this upstream as this is very brittle.

Reply:
A toolbar definition may indeed not be fully known by the time the open is triggered as some plugins may not yet be loaded or the treatment of the settings may not be complete yet. Hence adding some sort of delay does the trick.